lua/neotest-jest/init.lua

```diff
@@ -146,33 +153,27 @@ function adapter.build_spec(args)
     testNamePattern = "'" .. escapeTestPattern(pos.name:gsub("'", "")) .. "$'"
   end
 
-  local binary = getJestCommand(pos.path) or "jest"
```

Comment:
Yep! It's intentional. The `getJestCommand` has been annotated with a return type of `string` (not `string?` or `string|nil`) because it _always_ returns a string. Meaning the `or "jest"` never executes.